firebase-firestore/src/test/java/com/google/firebase/firestore/local/LocalStoreTestCase.java

```diff
@@ -934,4 +949,143 @@ public void testRemoteDocumentKeysForTarget() {
     keys = localStore.getRemoteDocumentKeys(2);
     assertSetEquals(asList(key("foo/bar"), key("foo/baz")), keys);
   }
+
+  @Test
+  public void testHandlesSetMutationThenTransformMutationThenTransformMutation() {
+    writeMutation(setMutation("foo/bar", map("sum", 0)));
+    assertContains(doc("foo/bar", 0, map("sum", 0), Document.DocumentState.LOCAL_MUTATIONS));
+    assertChanged(doc("foo/bar", 0, map("sum", 0), Document.DocumentState.LOCAL_MUTATIONS));
+
+    writeMutation(transformMutation("foo/bar", map("sum", FieldValue.numericAdd(1))));
```

Comment:
FWIW, the actual client will never write a batch with just a transform mutation. It'll always be preceded by a patch or set mutation...  So this test is slightly non-realistic. I don't care strongly though.